Add helm dda mapper to kubectl-plugin (#2215)
DataDog/datadog-operator: cmd/yaml-mapper/README.md

# README.md

## Purpose

The purpose of this tool is to map a YAML file of a certain structure to another YAML file of a different structure. For instance, migrating a Helm chart `values.yaml` file to another `values.yaml` file after a significant chart update.

## Motivation

> The motivation for creating this tool was to provide a way to support Datadog users who desire to switch from deploying the Datadog Agent using the `datadog` Helm chart to using the Datadog Operator controller. It is a potentially significant change that requires creating a new `DatadogAgent` custom resource specification. As a result, we are providing a way to map from a Helm chart `values.yaml` file to a `DatadogAgent` CRD spec, using a provided mapping.yaml file.

Reviewer: ```suggestion
The motivation for creating this tool is to provide a way to support Datadog users who want to switch from using the `datadog` Helm chart to using the Datadog Operator controller when deploying the Datadog Agent. It is a significant change that requires creating a new `DatadogAgent` custom resource specification. As a result, we are providing a way to map from a Helm chart `values.yaml` file to a `DatadogAgent` CRD spec, using a provided `mapping.yaml` file.
```

The resulting file is written to `dda.yaml.<timestamp>`. To specify a destination file, use flag `--destPath=[<FILENAME>.yaml]`.

Content from a file can be optionally prepended to the output. To specify the header file, use flag `--headerPath=[<FILENAME>.yaml]`.

By default, the output is also printed to STDOUT; to disable this use the flag `--printOutput=false`.

## Example usage

```bash
yaml-mapper --sourcePath=<EXAMPLE_SOURCE>.yaml --mappingPath=mapper/mapping_datadog_helm_to_datadogagent_crd.yaml --headerPath=<EXAMPLE_HEADER>.yaml
```

The following command writes the mapped DDA to the provided `destination.yaml` file. 
```bash
yaml-mapper --sourcePath=examples/example_source.yaml --mappingPath=mapper/mapping_datadog_helm_to_datadogagent_crd.yaml --headerPath=examples/example_header.yaml --destPath=examples/destination.yaml
```

### Update Mapping File from a Source YAML

*When updating the mapping file, please be sure to add the [corresponding key!](#updating-mapping-keys)*

Below are different ways to update the mapping file based on your source:

1. **Local values.yaml from your branch**

> If you have run into a CI error when adding a new field to values.yaml, run this command:

Reviewer: ```suggestion
If you have run into a CI error when adding a new field to `values.yaml`, run this command:
```
